backend/projects/views/dimension.py

```diff
@@ -0,0 +1,55 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from projects.models import DynamicDimension, DimensionMetaData, ProjectDimension
+from projects.permissions import IsProjectAdmin, IsProjectStaffAndReadOnly
+from projects.serializers import DynamicDimensionSerializer, DimensionMetaDataSerializer, ProjectDimensionSerializer
+
+
+class DimensionMetaDataList(generics.ListCreateAPIView):
+    serializer_class = DimensionMetaDataSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DimensionMetaData.objects.filter(dimension=self.kwargs["dimension_id"])
+
+
+class DynamicDimensionList(generics.ListCreateAPIView):
+    serializer_class = DynamicDimensionSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DynamicDimension.objects.all()
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class GetAllDynamicDimensions(generics.ListAPIView):
+    serializer_class = DynamicDimensionSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return DynamicDimension.objects.all()
+
+
+class AddProjectDimension(generics.CreateAPIView):
+    serializer_class = ProjectDimensionSerializer
+
+    def get_queryset(self):
+        return ProjectDimension.objects.all()
+
+    def perform_create(self, serializer):
+        project_id = self.kwargs["project_id"]
+        dimensions = self.request.data.get("dimension", [])
+
+        dimension_objs = DynamicDimension.objects.filter(pk__in=dimensions)
+        for dimension in dimension_objs:
+            project_dimension = ProjectDimension.objects.create(
+                project_id=project_id, dimension=dimension
+            )
```

Comment:
yeah, it is because of the unique constraint in Meta class of the model. It is not allowed to assign duplicated dimensions to the same project. So you will get the message of violate constraint. So I dont know what should I add here :sweat_smile: like a message or something?

Reply:
I was thinking maybe it can be put inside a try-catch block, where if the assigning of dimension1 in `dimension_objs` fails, it will just continue to dimension2 and so on. So if the project already has dims [1, 2, 3], the assigning of [1, 4, 5] will work. What do you think? 😅

Reply:
I think it's better to validate the selected dimension when they select it (for example when they choose 1 there'll be alert that 1 is already exist, pls remove). Because if we let the user pass the whole list of dimensions and let the 4, 5 assigning work silently even the 1 is violated, it's a bit unsafe :sweat_smile: How do u think?

Reply:
ofc this way I will give the endpoint to check the existance :sweat_smile:

Reply:
I see. I think it is reasonable argument. 😅 In this case, let's keep the code as it is now. The validation on FE side can be done because there is already an api for listing the dims of a particular project, which can be used for the purpose. I will mark this point as resolved 😅